### PR TITLE
E2E test: Add custom screenshots from slice page and changes page

### PR DESCRIPTION
**Pull request by mdeclercq**
## Context

➡️  https://linear.app/prismic/issue/SM-908/[spike]-[e2e-test]-screenshotv2-features

## The Solution
- Introduce the Page Object pattern for end-to-end testing to improve test readability: to be discussed, this may conflict with using custom commands.
- Compare images by comparing hashes of the base64 content. An easier alternative could be to only compare image dimensions but I'm not sure that's enough.
- Added 2 tests that can run [independently](https://docs.cypress.io/guides/references/best-practices#Having-tests-rely-on-the-state-of-previous-tests) (nominal + error case)
- Non-flakiness verification: I ran the CI 3 times in a row, e2e step was always successful. 
- Upload screenshots of Cypress tests failures as Github ci artifacts 

## Checklist before requesting a review
- [x] I hereby declare my code ready for review.
- [x] If it is a critical feature, I have added tests.
- [x] The CI is successful.
- [x] If there could backward compatibility issues, it has been discussed and planned.

### CODE REVIEW

File: cypress/helpers/images.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+// Compare an element src attribute to a fixture image, using a hash of the Base64 images
+export function isSameImageAs(subject, referenceImage) {
```
Comment (MarcMcIntosh) [inline cypress/helpers/images.js:2]: neat

File: cypress/helpers/slices.js [hunk 2]
```diff
@@ -112,6 +112,22 @@ function addFieldToSlice(selectors, fieldType, fieldName, fieldId) {
...
+ * On the Slice builder, add a new variation through the variation modal.
+ *
+ * @param {string} variationName Name of the variation.
+ */
+export function addVariationToSlice(variationName) {
+  cy.get("[aria-label='Expand variations']").click({ force: true });
+  cy.contains("button", "Add new variation").click();
+
+  cy.get("[aria-modal]").within(() => {
+    cy.get('input[name="name"]').type(variationName);
```
Comment (MarcMcIntosh) [inline cypress/helpers/slices.js]: Somewhere there is a function for selecting the input by label if that helps :)

File: cypress/pages/slices/slicePage.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
+export class SlicePage {
+  get saveButton() {
+    return cy.contains("Save to File System");
+  }
+  get imagePreview() {
+    return cy.get("[alt='Preview image']");
+  }
+
+  goTo(sliceName) {
+    cy.visit(`/slices/${sliceName}/default`);
```
Comment (MarcMcIntosh) [inline cypress/pages/slices/slicePage.js]: note for future refrence: the library name doesn't always need to be slices, it just usually is ;)

File: packages/slice-machine/lib/builders/SliceBuilder/Header/VariationsPopover/index.tsx [hunk 4]
```diff
@@ -88,9 +88,9 @@ const VarationsPopover: React.FunctionComponent<{
         >
           {current.name}{" "}
           {!isOpen ? (
-            <RiArrowDropDownFill size="16px" />
+            <RiArrowDropDownFill size="16px" aria-label="Expand variations" />
```
Comment (MarcMcIntosh) [inline packages/slice-machine/lib/builders/SliceBuilder/Header/VariationsPopover/index.tsx:91]: Good idea to aria-labels

File: .github/workflows/test-and-deploy-alpha.yml [hunk 5]
```diff
@@ -104,6 +104,11 @@ jobs:
           user_service_endpoint: ${{ secrets.USER_SERVICE_ENDPOINT }}
           acl_provider_endpoint: ${{ secrets.ACL_PROVIDER_ENDPOINT }}
         run: npm i && npm run test:e2e
+      - uses: actions/upload-artifact@v3
+        if: failure()
+        with:
+          name: e2e-tests-screenshots
+          path: cypress/screenshots/
```
Comment (MarcMcIntosh) [inline .github/workflows/test-and-deploy-alpha.yml:111]: nice

**Review by MarcMcIntosh — APPROVED** [review]

File: cypress/pages/slices/slicePage.js (hunk 3 above)
Comment (mdeclercq) [reply cypress/pages/slices/slicePage.js]: good point, I could add an additional parameter for the slices directory

File: cypress/helpers/slices.js (hunk 2 above)
Comment (mdeclercq) [reply cypress/helpers/slices.js]: I updated it, it's much better 🙏

File: cypress/pages/slices/slicePage.js (hunk 3 above)
Comment (mdeclercq) [reply cypress/pages/slices/slicePage.js]: done

Comment (mdeclercq) [reply cypress/pages/slices/slicePage.js]: done

**Review by MarcMcIntosh — APPROVED** [review]

**Pull request merged**